platform/security/sts/security-sts-attributequeryclaimshandler/src/main/java/org/codice/ddf/security/claims/attributequery/AttributeQueryClaimsHandler.java

```diff
@@ -0,0 +1,271 @@
+
+        //If security claim collection came back null, return a null claim collection.
+        if (securityClaimCollection == null || securityClaimCollection.size() == 0) {
+            return null;
+        }
+
+        claimCollection.addAll(securityClaimCollection);
+
+        return claimCollection;
+    }
+
+    /**
+     * Parses the CN from the user's DN.
+     *
+     * @param principal of the user.
+     * @return CN of the user.
+     */
+    protected String getUsername(Principal principal) {
+        String cn = null;
+        StringTokenizer st = new StringTokenizer(principal.getName(), ",");
+        while (st.hasMoreElements()) {
+            String[] strArr = st.nextToken().split("=");
+            if (strArr.length > 1 && strArr[0].equalsIgnoreCase("cn")) {
+                cn = strArr[1];
+                break;
+            }
+        }
+        return cn;
+    }
+
+    /**
+     * Gets the attributes for the supplied user from the external attribute store.
+     *
+     * @param username used for the request.
+     * @return The collection of attributes retrieved from the external attribute store.
+     * @throws URISyntaxException
+     */
+    protected ProcessedClaimCollection getAttributes(String username) throws URISyntaxException {
+        ProcessedClaimCollection claimCollection = new ProcessedClaimCollection();
+
+        LOGGER.debug("Sending AttributeQuery Request.");
+
+        Assertion assertion = null;
+        try {
+            assertion = createAttributeQueryClient(simpleSign, externalAttributeStoreUrl, issuer,
+                    destination).retrieveResponse(username);
+        } catch (AttributeQueryException ex) {
+            LOGGER.error("Error occurred in AttributeQueryClient, did not retrieve response.", ex);
+        }
+
+        if (assertion != null) {
+            createClaims(claimCollection, assertion);
+        }
+        return claimCollection;
+    }
+
+    /**
+     * Creates claims based on the retrieved attributes.
+     *
+     * @param claimsCollection The collection of claims.
+     * @param assertion        Assertion of the response.
+     * @return The collection of claims.
+     * @throws URISyntaxException
+     */
+    protected ProcessedClaimCollection createClaims(ProcessedClaimCollection claimsCollection,
+            Assertion assertion) throws URISyntaxException {
+
+        AttributeStatement attributeStatement = assertion.getAttributeStatements().get(0);
+        List<Attribute> attributeList = attributeStatement.getAttributes();
+
+        // If the claim is supported, check if it has an attribute mapping.
+        // If so, map the attribute (attribute value -> mapped attribute value)
+        // and create the claim, otherwise, create the claim using its original attribute value.
+        for (Attribute attribute : attributeList) {
+            for (String claimType : supportedClaims) {
```

Comment:
Then I would have the problem where if the attribute that is extracted from Assertion is the same but has a different letter case, I'd be creating the Claim using the wrong claim type, as I would be creating it using the attribute.getName() instead of the claimType from the list of supported claims.

So it's a matter of supporting letter case or not.